gulpfile.js

```diff
@@ -200,128 +200,128 @@ function clean_cache() {
+        './src/css/dropdown-lists/css/style_lists.css',
+        './src/css/font-awesome/css/font-awesome.min.css',
+        './src/js/libraries/flightindicators.css',
+        './src/js/libraries/jbox/jBox.css',
+        './src/js/libraries/jbox/themes/NoticeBorder.css',
+        './src/js/libraries/jbox/themes/ModalBorder.css',
+        './src/js/libraries/jbox/themes/TooltipDark.css',
+        './src/js/libraries/jbox/themes/TooltipBorder.css',
+        './src/js/libraries/jquery.nouislider.pips.min.css',
+        './src/js/libraries/switchery/switchery.css',
+        './src/js/libraries/jquery.nouislider.min.css',
 
         // JavaScript
-        './js/libraries/q.js',
-        './js/libraries/jquery-2.1.4.min.js',
-        './js/libraries/jquery-ui-1.11.4.min.js',
-        './js/libraries/d3.min.js',
-        './js/libraries/jquery.nouislider.all.min.js',
-        './js/libraries/three/three.min.js',
-        './js/libraries/three/Projector.js',
-        './js/libraries/three/CanvasRenderer.js',
-        './js/libraries/jquery.flightindicators.js',
-        './js/libraries/semver.js',
-        './js/libraries/jbox/jBox.min.js',
-        './js/libraries/switchery/switchery.js',
-        './js/libraries/bluebird.min.js',
-        './js/libraries/jquery.ba-throttle-debounce.min.js',
-        './js/libraries/inflection.min.js',
-        './js/injected_methods.js',
-        './js/data_storage.js',
-        './js/workers/hex_parser.js',
-        './js/fc.js',
-        './js/port_handler.js',
-        './js/port_usage.js',
-        './js/serial.js',
-        './js/gui.js',
-        './js/huffman.js',
-        './js/default_huffman_tree.js',
-        './js/model.js',
-        './js/serial_backend.js',
-        './js/msp/MSPCodes.js',
-        './js/msp.js',
-        './js/msp/MSPHelper.js',
-        './js/backup_restore.js',
-        './js/peripherals.js',
-        './js/protocols/stm32.js',
-        './js/protocols/stm32usbdfu.js',
-        './js/localization.js',
-        './js/boards.js',
-        './js/RateCurve.js',
-        './js/Features.js',
-        './js/Beepers.js',
-        './js/release_checker.js',
-        './tabs/adjustments.js',
-        './tabs/auxiliary.js',
-        './tabs/cli.js',
-        './tabs/configuration.js',
-        './tabs/failsafe.js',
-        './tabs/firmware_flasher.js',
-        './tabs/gps.js',
-        './tabs/help.js',
-        './tabs/landing.js',
-        './tabs/led_strip.js',
-        './tabs/logging.js',
-        './tabs/map.js',
-        './tabs/motors.js',
-        './tabs/onboard_logging.js',
-        './tabs/osd.js',
-        './tabs/pid_tuning.js',
-        './tabs/ports.js',
-        './tabs/power.js',
-        './tabs/receiver.js',
-        './tabs/receiver_msp.js',
-        './tabs/sensors.js',
-        './tabs/servos.js',
-        './tabs/setup.js',
-        './tabs/setup_osd.js',
-        './tabs/transponder.js',
-        './main.js',
+        './src/js/libraries/q.js',
```

Comment:
`libraries` should really go into a `lib/` folder or something, and not into the `src/` folder, since they are not sources of the project itself. The same goes for `css/fonts/`.

Reply:
@mikeller: I didn't touch the folder structure under the 'js' folder, this was already how it has been setup. As these are client libraries (running in the browser) this isn't a strange place to put them.

Reply:
I know you didn't. I always thought it was wrong and confusing to have a folder that contains a mix of files that we created and files that were copied from the internet, and have a `js/` folder, and `.js` files that live outside of this folder. I never changed it since before we used `gulp` and just zipped and shipped the entire repository to the Google web store, any new structure was still going to be a compromise.
Now that we have gulp, and use a build process to assemble the the application from parts, we should really clean this up and not just move a mess from one place to another.

Reply:
If you are not comfortable with doing these changes, we can just move all of the existing application files (without the changelog) into the `src/` folder for now, and do the cleanup in a later step. But I strongly believe that the cleanup should be done.

Reply:
Yes, no problem, now we have the hood open, just as good do it the write way.
Any other files you want to move around?

Reply:
I think that's it for now - if we find stuff later that we are not happy with we can move it then. Doing it now will also give us some time for bugs to be found if the pathing is not correct after the move, or we forgot something.